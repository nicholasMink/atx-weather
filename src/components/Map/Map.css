.mapWrapper {
  height: 200px;
  width: 100%;
  background: #eee;
  color: #aaa;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0;
}
.tipText {
  background: rgba(0, 0, 0, .7);
  width: 10rem;
  color: #fff;
  font-size: 1rem;
  padding: 1rem;
  border: .13rem solid #fff;
}