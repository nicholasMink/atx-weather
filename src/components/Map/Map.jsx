import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';
import './Map.css';

const TipText = ({ text }) => <div className="tipText">{text}</div>;

export default class Map extends Component {
  static defaultProps = {
    center: {
      lat: 30.27,
      lng: -97.745
    },
    zoom: 11
  };
  render() {
    return (
      // Important! Always set the container height explicitly
      <div style={{ height: '100vh', width: '100%' }} >
        <GoogleMapReact bootstrapURLKeys={{ key: 'AIzaSyDlRnBdMvhFPtUdJnramc_kLQOwMEyfbYs' }} defaultCenter={this.props.center} defaultZoom={this.props.zoom}>
          <TipText lat={30.231798} lng={-97.780521} text={'MINK Weather Station'} />
        </GoogleMapReact>
      </div>
    )
  }
}
