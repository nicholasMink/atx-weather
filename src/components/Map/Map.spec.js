import React from 'react';
import {shallow} from 'enzyme';
import GoogleMapReact from 'google-map-react';
import Map, { TipText } from './Map';

describe('Map', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Map />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 1 <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  // it('should render the GoogleMapReact Component', () => {
  //   expect(wrapper.containsMatchingElement(<GoogleMapReact />)).toEqual(true);
  // });

  // it('should render the TipText Component', () => {
  //   expect(wrapper.containsMatchingElement(<TipText />)).toEqual(true);
  // });
});