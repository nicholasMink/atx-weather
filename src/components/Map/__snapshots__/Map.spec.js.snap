// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    center={
        Object {
            "lat": 30.27,
            "lng": -97.745,
          }
    }
    zoom={11}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GoogleMap
        bootstrapURLKeys={
                Object {
                        "key": "AIzaSyDlRnBdMvhFPtUdJnramc_kLQOwMEyfbYs",
                      }
        }
        debounced={true}
        defaultCenter={
                Object {
                        "lat": 30.27,
                        "lng": -97.745,
                      }
        }
        defaultZoom={11}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        style={
                Object {
                        "height": "100%",
                        "margin": 0,
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
        }
        yesIWantToUseGoogleMapApiInternals={false}
>
        <Unknown
                lat={30.231798}
                lng={-97.780521}
                text="MINK Weather Station"
        />
</GoogleMap>,
      "style": Object {
        "height": "100vh",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bootstrapURLKeys": Object {
          "key": "AIzaSyDlRnBdMvhFPtUdJnramc_kLQOwMEyfbYs",
        },
        "children": <Unknown
          lat={30.231798}
          lng={-97.780521}
          text="MINK Weather Station"
/>,
        "debounced": true,
        "defaultCenter": Object {
          "lat": 30.27,
          "lng": -97.745,
        },
        "defaultZoom": 11,
        "distanceToMouse": [Function],
        "googleMapLoader": [Function],
        "heatmap": Object {},
        "heatmapLibrary": false,
        "hoverDistance": 30,
        "layerTypes": Array [],
        "options": [Function],
        "style": Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        },
        "yesIWantToUseGoogleMapApiInternals": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "lat": 30.231798,
          "lng": -97.780521,
          "text": "MINK Weather Station",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GoogleMap
          bootstrapURLKeys={
                    Object {
                              "key": "AIzaSyDlRnBdMvhFPtUdJnramc_kLQOwMEyfbYs",
                            }
          }
          debounced={true}
          defaultCenter={
                    Object {
                              "lat": 30.27,
                              "lng": -97.745,
                            }
          }
          defaultZoom={11}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          heatmap={Object {}}
          heatmapLibrary={false}
          hoverDistance={30}
          layerTypes={Array []}
          options={[Function]}
          style={
                    Object {
                              "height": "100%",
                              "margin": 0,
                              "padding": 0,
                              "position": "relative",
                              "width": "100%",
                            }
          }
          yesIWantToUseGoogleMapApiInternals={false}
>
          <Unknown
                    lat={30.231798}
                    lng={-97.780521}
                    text="MINK Weather Station"
          />
</GoogleMap>,
        "style": Object {
          "height": "100vh",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bootstrapURLKeys": Object {
            "key": "AIzaSyDlRnBdMvhFPtUdJnramc_kLQOwMEyfbYs",
          },
          "children": <Unknown
            lat={30.231798}
            lng={-97.780521}
            text="MINK Weather Station"
/>,
          "debounced": true,
          "defaultCenter": Object {
            "lat": 30.27,
            "lng": -97.745,
          },
          "defaultZoom": 11,
          "distanceToMouse": [Function],
          "googleMapLoader": [Function],
          "heatmap": Object {},
          "heatmapLibrary": false,
          "hoverDistance": 30,
          "layerTypes": Array [],
          "options": [Function],
          "style": Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          },
          "yesIWantToUseGoogleMapApiInternals": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "lat": 30.231798,
            "lng": -97.780521,
            "text": "MINK Weather Station",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
